packages/asset-service/src/service/AssetService.ts

```diff
@@ -1,9 +1,64 @@
+import { caip19 } from '@shapeshiftoss/caip'
 import { Asset, AssetDataSource, BaseAsset, ChainTypes, NetworkTypes } from '@shapeshiftoss/types'
 import axios from 'axios'
 
+/* eslint-disable */
+const Identicon = require('identicon.js')
+const { fonts, renderPixels } = require('js-pixel-fonts')
+/* eslint-enable */
+
 import assetsDescriptions from './descriptions.json'
 import localAssetData from './generatedAssetData.json'
 
+const getRenderedIdenticonB64 = (symbol: string, tokenId: string): string => {
```

Comment:
I think the original demand was to have a module, in case you want to generate the image:

`Add a module to asset-service to deterministically generate an icon based on the caip19 of an asset.`

It means that this method should be exported from `asset-service` so any app using it can generate the icon on the fly

`During rendering in web, if the asset does not have a remote resource icon (property icon on the Asset type), use the new module to generate the icon on the fly and display that icon.`

I love the fact that finally, we provide directly the generated icon into the asset, instead of making every users generate it but it can lead to some places where you want to have a 64x64 icon but you only have 128x128, wdyt?

Reply:
You are right, I have updated the PR to allow more configurable options.